tsfc/fem.py

```diff
@@ -697,6 +697,15 @@ def compile_ufl(expression, interior_facet=False, point_sum=False, **kwargs):
 
     # Translate UFL to GEM, lowering finite element specific nodes
     result = map_expr_dags(context.translator, expressions)
+    # Check that any indices from the PointSet have made it into the resulting expression
+    import functools
+    import operator
+    for i in range(len(result)):
+        expansion_indices = tuple(j for j in context.point_indices if j not in result[i].free_indices)
+        # blow up shape to include expansion_indices using deltas - these will
+        # eventually be cancelled out but ensure the our result has the expected
+        # free_indices and shape.
+        result[i] = functools.reduce(operator.mul, (gem.Delta(j, j) for j in expansion_indices), result[i])
```

Comment:
Given that this loop always produces new results:

```python
result = [functools.reduce(operator.mul, (gem.Delta(j, j)
                                          for j in context.point_indices
                                          if j not in expr.free_indices),
                           expr)
          for expr in result]
```

Makes it clearer that we're modifying every result entry.

Reply:
Nice. Done in d711e722d19ce86ae5fca8b60e01492bd55e6ae6